### PR TITLE
Pull request just for initial review purposes

### PR DESCRIPTION
**Pull request by dpwspoon**

### CODE REVIEW

File: management/pom.xml [hunk 1]
```diff
@@ -148,6 +148,11 @@
             <artifactId>snmp4j</artifactId>
         </dependency>
         <dependency>
+            <groupId>uk.co.real-logic</groupId>
+            <artifactId>Agrona</artifactId>
+            <version>0.3.1</version>
```
Comment (dpwspoon) [inline management/pom.xml]: Move this to dependencyManagement section of top level pom.  And then remove the version

File: management/src/main/java/org/kaazing/gateway/management/context/ManagementContext.java [hunk 2]
```diff
@@ -43,7 +43,7 @@
 import org.kaazing.gateway.util.scheduler.SchedulerProvider;
 import org.kaazing.mina.core.session.IoSessionEx;
 
-public interface ManagementContext {
+public interface ManagementContext extends AutoCloseable {
```
Comment (dpwspoon) [inline management/src/main/java/org/kaazing/gateway/management/context/ManagementContext.java]: This isn't really AutoCloseable, so suggest changing interface directly

File: management/src/main/java/org/kaazing/gateway/management/monitoring/entity/factory/MonitoringEntityFactory.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+/**
+ * Factory for instantiating a specific monitoring entity.
+ *
+ * Monitoring entities represent values which indicate specific gateway relevant information, such as the number of open
+ * sessions, the number of bytes sent/received, etc.
+ *
+ * This interface exposes the API for retrieving monitoring entities.
+ */
+public interface MonitoringEntityFactory extends AutoCloseable {
```
Comment (dpwspoon) [inline management/src/main/java/org/kaazing/gateway/management/monitoring/entity/factory/MonitoringEntityFactory.java]: Suggest putting close method directly on the MonitoringEntityFactory interface

File: management/src/main/java/org/kaazing/gateway/management/monitoring/entity/impl/DefaultLongMonitoringCounterStub.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Override
+    public LongMonitoringCounter increment() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public LongMonitoringCounter incrementByValue(long value) {
+        // TODO Auto-generated method stub
```
Comment (dpwspoon) [inline management/src/main/java/org/kaazing/gateway/management/monitoring/entity/impl/DefaultLongMonitoringCounterStub.java]: Remove auto generated TODO, put in a real comment if needed

File: management/src/test/mock/LongMonitoringCounterMock.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.kaazing.gateway.management.monitoring.entity.mock;
+
+import org.kaazing.gateway.management.monitoring.entity.LongMonitoringCounter;
+
```
Comment (dpwspoon) [inline management/src/test/mock/LongMonitoringCounterMock.java]: Remove this class

**Comment by dpwspoon** [comment]
Should add some unit tests

**Comment by dpwspoon** [comment]
See [my branch](https://github.com/dpwspoon/gateway/tree/management) for cleaning up on the gateway stopped

File: management/src/main/java/org/kaazing/gateway/management/monitoring/configuration/impl/DefaultMonitoringEntityFactoryBuilderStub.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+import org.kaazing.gateway.management.monitoring.configuration.MonitoringEntityFactoryBuilder;
+import org.kaazing.gateway.management.monitoring.entity.factory.MonitoringEntityFactory;
+import org.kaazing.gateway.management.monitoring.entity.impl.DefaultMonitoringEntityFactoryStub;
+
+public class DefaultMonitoringEntityFactoryBuilderStub implements MonitoringEntityFactoryBuilder {
+
+    @Override
+    public MonitoringEntityFactory build() {
+        // TODO Auto-generated method stub
```
Comment (brennangaunce) [inline management/src/main/java/org/kaazing/gateway/management/monitoring/configuration/impl/DefaultMonitoringEntityFactoryBuilderStub.java]: Remove todo.

File: management/src/main/java/org/kaazing/gateway/management/context/DefaultManagementContext.java [hunk 7]
```diff
@@ -59,6 +64,10 @@
 import org.kaazing.gateway.management.gateway.GatewayManagementBeanImpl;
 import org.kaazing.gateway.management.gateway.GatewayManagementListener;
 import org.kaazing.gateway.management.gateway.ManagementGatewayStrategy;
+import org.kaazing.gateway.management.monitoring.configuration.MonitoringEntityFactoryBuilder;
+import org.kaazing.gateway.management.monitoring.configuration.impl.AgronaMonitoringEntityFactoryBuilder;
```
Comment (brennangaunce) [inline management/src/main/java/org/kaazing/gateway/management/context/DefaultManagementContext.java]: impl classes should not be imported at this level.  Should use a factory for this.

**Comment by dpwspoon** [comment]
@alingherasim can we close this?

**Comment by alingherasim** [comment]
Yes, we can close this pull request. We will create a new pull request which will contain the necessary functionality.

**Pull request closed without merging**